snapcraft/pluginhandler.py

```diff
@@ -223,11 +223,13 @@ def get_state(self, step):
     def _step_state_file(self, step):
         return os.path.join(self.statedir, step)
 
-    def _setup_stage_packages(self):
-        ubuntu = repo.Ubuntu(
+    def _fetch_stage_packages(self):
+        self.ubuntu = repo.Ubuntu(
```

Comment:
Good catch, not sure what I was thinking there. I ended up extracting it into a property instead that initializes itself on first use (putting it in `__init__` would make it fetch sources every time a step was run, which wouldn't be ideal).